version: '3.4'

services:
  db:
    restart: always
    container_name: voltgo_db
    image: postgres:14.9-alpine
    volumes:
      - db:/var/lib/postgresql/data
    networks:
      - voltgo
    environment:
      - POSTGRES_PASSWORD=postgres
  web:
    restart: always
    container_name: voltgo_web
    image: voltgo_web:latest
    build: .
    environment:
      - STRIPE_PUBLIC_KEY=pk_test_51OHWDGGXSBM20rPRZ6SWDsMZgnA5KXRfuSvG5V7yoCWL90qFiLalG3ggssj6rvynDmx6JB49CG3F1kLQmGeZFxkD00zga40NBD
      - STRIPE_SECRET_KEY=sk_test_51OHWDGGXSBM20rPRxhKaoXCzDbXklKvuYZJqrIsQkpg1bq9Voc94E4OReS33BriBiyIAEDE8nCg5E2yvuMATNc5V000rY2IUn4
    command: ash -c "python manage.py migrate && gunicorn -w 5 voltgo.wsgi --timeout=500 -b 0.0.0.0:5000"
    expose:
      - "5000"
    volumes:
      - static:/app/static
    depends_on:
      - db
    networks:
      - voltgo

volumes:
  static:
    name: voltgo_static
  db:
    name: voltgo_db

networks:
  voltgo:
    driver: bridge
    ipam:
     driver: default
     config:
       - subnet: 10.5.0.0/16